backend/de.metas.adempiere.adempiere/base/src/main/java/de/metas/comments/CommentEntryRepository.java

```diff
@@ -131,10 +161,45 @@ private CommentEntryParentId getOrCreateParent(final @NonNull TableRecordReferen
 	private CommentEntryParentId getParentIdOrNull(final @NonNull TableRecordReference tableRecordReference)
 	{
 		return queryBL.createQueryBuilder(I_CM_Chat.class)
+				.addOnlyActiveRecordsFilter()
 				.addEqualsFilter(I_CM_Chat.COLUMNNAME_AD_Table_ID, tableRecordReference.getAD_Table_ID())
 				.addEqualsFilter(I_CM_Chat.COLUMNNAME_Record_ID, tableRecordReference.getRecord_ID())
 				.orderBy(I_CM_Chat.COLUMNNAME_CM_Chat_ID)
 				.create()
 				.firstId(CommentEntryParentId::ofRepoIdOrNull);
 	}
+
+	@NonNull
+	private Map<TableRecordReference, Boolean> checkReferenceHasComments(@NonNull final Collection<TableRecordReference> referencesUnknownStatus)
+	{
+		final ImmutableListMultimap<Integer, Integer> tablesForRecords = referencesUnknownStatus.stream()
+				.collect(ImmutableListMultimap.toImmutableListMultimap(TableRecordReference::getAD_Table_ID, TableRecordReference::getRecord_ID));
+
+		// first query returns nothing. Rest of the queries return the correct data
+		final IQuery<I_CM_Chat> query = queryBL.createQueryBuilder(I_CM_Chat.class)
+				.addEqualsFilter(I_CM_Chat.COLUMNNAME_CM_Chat_ID, -1)
```

Comment:
but why?

Reply:
:(((

Reply:
Teo suggested to use `.reduce(IQuery.unionDistict())`.
TIL